Does not build on Windows Server 2016

Turned up a fresh EC2 on AWS with Windows Server 2016, followed the [Getting Started](https://github.com/microsoft/ebpf-for-windows/blob/master/docs/GettingStarted.md) procedure verbatim. Everything looked good until the `msbuild` step. First issue was a dialog that said:
```
The procedure entry point VirtualAlloc2 could not be located in the dynamic link library C:\Users\Administrator\git\ebpf-for-windows\x64\Debug\encode_program_info.exe
```
Then the build failed after dismissing the dialog with:
```
C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(439,5): error MSB8020: The build tools for WindowsKernelModeDriver10.0 (Platform Toolset = 'WindowsKernelModeDriver10.0') cannot be found. To build using the WindowsKernelModeDriver10.0 build tools, please install WindowsKernelModeDriver10.0 build tools.  Alternatively, you may upgrade to the current Visual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Retarget solution". [C:\Users\Administrator\git\ebpf-for-windows\libs\ubpf\kernel\ubpf_kernel.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(439,5): error MSB8020: The build tools for WindowsKernelModeDriver10.0 (Platform Toolset = 'WindowsKernelModeDriver10.0') cannot be found. To build using the WindowsKernelModeDriver10.0 build tools, please install WindowsKernelModeDriver10.0 build tools.  Alternatively, you may upgrade to the current Visual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Retarget solution". [C:\Users\Administrator\git\ebpf-for-windows\libs\execution_context\kernel\execution_context_kernel.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(286,5): error MSB3073: The command "cd /d C:\Users\Administrator\git\ebpf-for-windows\x64\Debug\ [C:\Users\Administrator\git\ebpf-for-windows\tools\encode_program_info\encode_program_info.vcxproj]C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(286,5): error MSB3073: C:\Users\Administrator\git\ebpf-for-windows\x64\Debug\encode_program_info.exe [C:\Users\Administrator\git\ebpf-for-windows\tools\encode_program_info\encode_program_info.vcxproj] C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(286,5): error MSB3073: :VCEnd" exited with code -1073741511. [C:\Users\Administrator\git\ebpf-for-windows\tools\encode_program_info\encode_program_info.vcxproj]
```
Maybe Windows Server 2016 isn't supported with this procedure?


Visual Studio details:
```
Microsoft Visual Studio Community 2019
Version 16.11.8
VisualStudio.16.Release/16.11.8+32002.261
Microsoft .NET Framework
Version 4.8.03761

Installed Version: Community

Visual C++ 2019   00435-60000-00000-AA081
Microsoft Visual C++ 2019

ASP.NET and Web Tools 2019   16.11.94.52318
ASP.NET and Web Tools 2019

Azure App Service Tools v3.0.0   16.11.94.52318
Azure App Service Tools v3.0.0

C# Tools   3.11.0-4.21403.6+ae1fff344d46976624e68ae17164e0607ab68b10
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Debugging Tools for Windows   10.0.19030.1000
Integrates the Windows Debugger functionality (http://go.microsoft.com/fwlink/?linkid=223405) in Visual Studio.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

NuGet Package Manager   5.11.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Enables Visual Studio's testing tools with unit tests written for Google Test.  The use terms and Third Party Notices are available in the extension installation directory.

TypeScript Tools   16.0.30526.2002
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.11.0-4.21403.6+ae1fff344d46976624e68ae17164e0607ab68b10
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Tools for CMake   1.0
Visual Studio Tools for CMake

Windows Driver Kit   10.0.19030.1000
Headers, libraries, and tools needed to develop, debug, and test Windows drivers (msdn.microsoft.com/en-us/windows/hardware/gg487428.aspx)
```

--- a/README.md
+++ b/README.md
@@ -37,7 +37,7 @@ Many helpers already exist, and more hooks and helpers will be added over time.
 
 ## Getting Started
 
-This project supports eBPF on Windows 10, and on Windows Server 2016 or later.
+This project supports eBPF on Windows 10 or later, and on Windows Server 2019 or later.
 To try out this project, see our [Getting Started Guide](docs/GettingStarted.md).
 
 Want to help?  We welcome contributions!  See our [Contributing guidelines](CONTRIBUTING.md).

--- a/docs/GettingStarted.md
+++ b/docs/GettingStarted.md
@@ -5,7 +5,8 @@
 The following must be installed in order to build this project:
 
 1. Git (e.g., [Git for Windows 64-bit](https://git-scm.com/download/win))
-2. [Visual Studio 2019 version 16.11.7 or later](https://visualstudio.microsoft.com/vs/), including
+2. [Visual Studio 2019 version 16.11.7 or later](https://www.techspot.com/downloads/downloadnow/7241/?evp=70f51271955e6392571f575e301cd9a3&file=9642), including
+   the "Desktop development with C++" workload, and
    the "MSVC v142 - VS 2019 C++ x64/x86 Spectre-mitigated libs (latest)"
    which must be selected as an Individual component in the VS installer
 3. [Visual Studio Build Tools 2019](https://aka.ms/vs/16/release/vs_buildtools.exe)
@@ -18,7 +19,7 @@ You should add the paths to `git.exe`, `cmake.exe` and `nuget.exe` to the Window
 ## How to clone and build the project
 This section outlines the steps to build, prepare and build the eBPF-For-Windows project.
 
-### Cloning the Project
+### Cloning the project
 1. ```git clone --recurse-submodules https://github.com/microsoft/ebpf-for-windows.git```.
 By default this will clone the project under the `ebpf-for-windows` directory.
 
@@ -31,8 +32,9 @@ The following steps need to be executed _once_ before the first build on a new c
 5. ```del external\ebpf-verifier\build\obj\project.assets.json```
 
 ### Building using Developer Command Prompt for VS 2019
-1. `"C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\Common7\Tools\VsDevCmd.bat"`
-2. ```msbuild /m /p:Configuration=Debug /p:Platform=x64 ebpf-for-windows.sln```
+1. Launch `Developer Command Prompt for VS 2019`.
+2. Change directory to where the project is cloned, e.g. ```cd ebpf-for-windows```.
+3. ```msbuild /m /p:Configuration=Debug /p:Platform=x64 ebpf-for-windows.sln```
 
 ### Building using Visual Studio IDE
 1. Open `ebpf-for-windows.sln`

